modules/sdk-core/src/account-lib/mpc/tss/eddsa/eddsa.ts

```diff
@@ -302,7 +309,28 @@ export default class Eddsa {
     return resultShares;
   }
 
-  sign(message: Buffer, playerShare: XShare, rShares: RShare[], yShares: YShare[] = []): GShare {
+  sign(
+    message: Buffer,
+    playerShare: XShare,
+    rShares: RShare[],
+    yShares: YShare[] = [],
+    commitments?: Commitment[]
+  ): GShare {
+    if (commitments) {
+      if (commitments.length !== rShares.length) {
+        throw new Error('Number of commitments must match number of rShares');
+      }
+      for (const commitment of commitments) {
+        const i_index = commitment.i;
+        const j_index = commitment.j;
+        const rShare = rShares.find((rShare) => rShare.i === i_index && rShare.j === j_index);
+        if (!rShare) {
+          throw new Error(`Could not find rShare for commitment ${commitment}`);
+        }
+        this.validateCommitment(rShare, commitment);
+      }
+    }
```

Comment:
I think it'll be better if we move the commitment to be a part of the R share. That way you don't have to do all this logic of trying to map the commitments 1:1 to the R-share and it's more intuitive.